Lib/typing.py

```diff
@@ -866,6 +870,75 @@ def is_str(val: Union[str, float]):
     return _GenericAlias(self, (item,))
 
 
+@_SpecialForm
+def TypeIs(self, parameters):
+    """Special typing construct for marking user-defined type narrowing functions.
+
+    ``TypeIs`` can be used to annotate the return type of a user-defined
+    type guard function.  ``TypeIs`` only accepts a single type argument.
+    At runtime, functions marked this way should return a boolean and accept
+    at least one argument.
+
+    ``TypeIs`` aims to benefit *type narrowing* -- a technique used by static
+    type checkers to determine a more precise type of an expression within a
+    program's code flow.  Usually type narrowing is done by analyzing
+    conditional code flow and applying the narrowing to a block of code.  The
+    conditional expression here is sometimes referred to as a "type guard".
```

Comment:
I'm a bit concerned about using "type guard" in this generic way (and similarly "type guard function" above) as a generic term, because it might tend to point toward `TypeGuard` as the default or canonical choice.

I don't think "type guard" is really standardized terminology to a degree that requires us to use it here.

Given that we are explicitly defining the term, I don't think "type predicate" is too jargon-y; what do you think?

It's also arguably more accurate: a "guard" would not just be an expression, it would be an entire `if` test IMO; the function itself is just a predicate.

Reply:
Agree, I will switch to "type predicate" consistently.